ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -96,7 +95,7 @@
     this.callback = callback;
     this.time = time;
     this.deletionTimeMs = time.milliseconds();
-    this.deleteRequestInfos = new HashMap<Integer, DeleteRequestInfo>();
+    this.deleteRequestInfos = new TreeMap<Integer, DeleteRequestInfo>();
```

Comment:
Why use treemap here?

Reply:
In fetchRequests function, this map will be iterated based on the order of it's key. So I change it to a sorted tree.

Reply:
how does that affect functionality?

Reply:
Since DeleteRequestInfo has a creationtTime, and the bigger the key is in the map, the latter the creationTime is in the request. We can rely on this correlation while trying to find all the timed out requests. As long as we hit a request that is not timed out, we can be sure that all the requests following that is not timed out.